src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -816,6 +818,81 @@ class spell_mage_flame_on : public AuraScript
    }
 };
 
+class FlurryEvent : public BasicEvent
+{
+    public:
+        FlurryEvent(Unit* caster, ObjectGuid const& targetGUID, uint8 count) : _caster(caster), _targetGUID(targetGUID), _count(count) { }
+
+        bool Execute(uint64 time, uint32 /*diff*/) override
+        {
+            Unit* target = ObjectAccessor::GetUnit(*_caster, _targetGUID);
+
+            if (!target)
+                return true;
+
+            _caster->CastSpell(target, SPELL_MAGE_FLURRY_DAMAGE, true);
+
+            if (!--_count)
+                return true;
+
+            _caster->m_Events.AddEvent(this, Milliseconds(time) + randtime(200ms, 300ms));
+            return false;
+        }
+
+    private:
+        Unit* _caster;
+        ObjectGuid _targetGUID;
+        uint8 _count;
+};
+
+// 44614 - Flurry
+class spell_mage_flurry : public SpellScript
+{
+    PrepareSpellScript(spell_mage_flurry);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_MAGE_FLURRY_DAMAGE
+        });
+    }
+
+    void EffectHit(SpellEffIndex)
+    {
+        GetCaster()->m_Events.AddEventAtOffset(new FlurryEvent(GetCaster(), GetHitUnit()->GetGUID(), GetEffectValue() - 1), randtime(200ms, 300ms));
```

Comment:
milliseconds like this usually imply that such events should happen alongside the main proc so better remove it alltogether and just move it into the effect handler

Reply:
No, EFFECT_1 of this spell is a SPELL_EFFECT_TRIGGER_MISSILE which also deals damage

Reply:
So why not attaching that part to the missile then?